eventrouter.h

```diff
@@ -124,6 +125,36 @@ extern "C"
     /// NOT be called from an interrupt or a callback.
     void ErUnsubscribe(ErModule_t *a_module, ErEventType_t a_event_type);
 
+    //============================================================================
+    // Implementation-Specific Functions
+    //============================================================================
+
+#ifdef ER_CONFIG_OS
+    /// Blocks until the next event sent to the current task is received, and
+    /// returns it; asserts if called from an interrupt.
+    ErEvent_t *ErReceive(void);
+
+    /// Blocks until either the next event sent to the current task is received
+    /// or `a_ms` milliseconds have passed. This function returns NULL on
+    /// timeout and asserts if called from an interrupt.
+    ErEvent_t *ErTimedReceive(int64_t a_ms);
+#endif
```

Comment:
These functions abstract the action reading events sent to the current task. They are necessary to make task logic that can be shared between FreeRTOS and POSIX applications.